mindsdb/interfaces/agents/langchain_agent.py

```diff
@@ -95,42 +92,37 @@ def get_llm_provider(args: Dict) -> str:
 def get_embedding_model_provider(args: Dict) -> str:
     """Get the embedding model provider from args.
 
-    For VLLM, this will return 'openai' since VLLM uses the OpenAI embeddings interface.
+    For VLLM, this will use our custom VLLMEmbeddings class from langchain_embedding_handler.
     """
+    # Check for explicit embedding model provider
     if "embedding_model_provider" in args:
         provider = args["embedding_model_provider"]
-        # If using VLLM embeddings, ensure base_url is provided
         if provider == 'vllm':
-            if not args.get('embedding_base_url'):
+            if not (args.get('openai_api_base') and args.get('model')):
```

Comment:
better to keep things consistent with openai interface, thus use same param names